src/libgeds/MetadataService.h

```diff
@@ -82,11 +89,33 @@ class MetadataService {
   absl::StatusOr<std::pair<std::vector<geds::Object>, std::vector<std::string>>>
   listPrefix(const std::string &bucket, const std::string &keyPrefix, char delimiter);
 
+  /**
+   * @brief List objects from cache in `bucket` starting with `key` as prefix. Objects that contain
+   * `delimiter` in the postfix of the key are filtered. Delimiter `\0` is treated as no filter.
+   */
+  absl::StatusOr<std::pair<std::vector<geds::Object>, std::vector<std::string>>>
+  listPrefixFromCache(const std::string &bucket, const std::string &keyPrefix, char delimiter);
+
   /**
    * @brief Prefix search with `/` as delimiter.
    */
   absl::StatusOr<std::pair<std::vector<geds::Object>, std::vector<std::string>>>
   listFolder(const std::string &bucket, const std::string &keyPrefix);
+
+  /**
+   * @brief Create subscription stream for the subscriber. This has to be called in a thread.
+   * The thread will be running for ever until it is stopped by
+   * setSubscribeStreamContinueAbortThreadFlag (false). The subscribeStreamContinueThreadFlag must
+   * be set first to true by setSubscribeStreamContinueAbortThreadFlag(true), before calling the
+   * method.
+   */
+  absl::Status subscribeStream(const geds::SubscriptionEvent &event);
+  absl::Status setSubscribeStreamContinueAbortThreadFlag(bool threadFlag);
```

Comment:
GEDS should always create a thread where it receives subscription events.
I would subscribe all events to the same subscriber ID. This way we avoid creating multiple threads waiting for different notifications.